contracts/CandidateHub.template

```diff
@@ -154,13 +154,13 @@ contract CandidateHub is ICandidateHub, System, IParamSubscriber {
           b. transfer the candidate's margin eth value to the SystemReward contract
   */    
   function jailValidator(address operateAddress, uint256 round, uint256 fine) 
-        external override onlyValidator onlyIfCandidateAddress {
+        external override onlyValidator onlyIfCandidateAddress(operateAddress) {
     uint256 index = operateMap[operateAddress];
     if (index == 0) return;
 
     Candidate storage c = candidateSet[index - 1];
     uint256 margin = c.margin;
-    if (margin >= dues && margin - dues >= fine) {
+    if (margin >= dues && margin - dues >= fine) { //@openissue
```

Comment:
margin  >=  dues + fine

Reply:
correct, you need this check if you subtract margin-duos. 
margin >= dues + fine  performs the exact same check with no danger of overflow